service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/FindResourcesToCloneStep.java

```diff
@@ -28,6 +28,13 @@
  * (cloneable) resource and possibly the flightID.
  */
 public class FindResourcesToCloneStep implements Step {
+  private static final List<StewardshipType> stewardshipCloneOrder =
+      List.of(StewardshipType.REFERENCED, StewardshipType.CONTROLLED);
+  private static final List<WsmResourceType> resourceCloneOrder =
+      List.of(
+          WsmResourceType.CONTROLLED_AZURE_STORAGE_CONTAINER,
+          WsmResourceType.CONTROLLED_AZURE_MANAGED_IDENTITY,
+          WsmResourceType.CONTROLLED_AZURE_DATABASE);
```

Comment:
here we enforce the clone order, so that we can ensure that the managed identity is available before the database clone process (so that it can be assigned as the database owner)

storage container is specified first as a holdover from our testing, but we leave it here because we think storage container cloning will eventually be moved from Rawls to WSM (based on discussion with @dvoet: [slack](https://broadinstitute.slack.com/archives/C02LMB1PQ6R/p1700673275677939))

resources not specified here will be cloned *before* these three resources.